kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/client/DefaultJsonHandlerSuite.scala

```diff
@@ -15,18 +15,204 @@
+    )
+  }
+
+  test("parse short type") {
+    testJsonParserForSingleType(
+      jsonString = """{"col1":-32767,"col2":8,"col3":32767}""",
+      dataType = ShortType.SHORT,
+      3,
+      TestRow(-32767.toShort, 8.toShort, 32767.toShort)
+    )
+  }
+
+  test("parse integer type") {
+    testJsonParserForSingleType(
+      jsonString = """{"col1":-2147483648,"col2":8,"col3":2147483647}""",
+      dataType = IntegerType.INTEGER,
+      3,
+      TestRow(-2147483648, 8, 2147483647)
+    )
+  }
+
+  test("parse long type") {
+    testJsonParserForSingleType(
+      jsonString = """{"col1":-9223372036854775808,"col2":8,"col3":9223372036854775807}""",
+      dataType = LongType.LONG,
+      3,
+      TestRow(-9223372036854775808L, 8L, 9223372036854775807L)
+    )
+  }
+
+  test("parse float type") {
+    testJsonParserForSingleType(
+      jsonString =
+      """{"col1":-9223.33,"col2":0.4,"col3":1.2E8,"col4":1.23E-7,"col5":0.004444444}""",
+      dataType = FloatType.FLOAT,
+      5,
+      TestRow(-9223.33F, 0.4F, 120000000.0F, 0.000000123F, 0.004444444F)
+    )
+  }
+
+  test("parse double type") {
+    testJsonParserForSingleType(
+      jsonString =
+        """
+          |{"col1":-9.2233333333E8,"col2":0.4,"col3":1.2E8,
+          |"col4":1.234444444E-7,"col5":0.0444444444}""".stripMargin,
+      dataType = DoubleType.DOUBLE,
+      5,
+      TestRow(-922333333.33D, 0.4D, 120000000.0D, 0.0000001234444444D, 0.0444444444D)
+    )
+  }
+
+  test("parse string type") {
+    testJsonParserForSingleType(
+      jsonString = """{"col1": "foo", "col2": "", "col3": null}""",
+      dataType = StringType.STRING,
+      3,
+      TestRow("foo", "", null)
+    )
+  }
+
+  test("parse decimal type") {
+    testJsonParserWithSchema(
+      jsonString = """
+      |{
+      |  "col1":0,
+      |  "col2":0.01234567891234567891234567891234567890,
+      |  "col3":123456789123456789123456789123456789,
+      |  "col4":1234567891234567891234567891.2345678900,
+      |  "col5":1.23
+      |}
+      |""".stripMargin,
+      schema = new StructType()
+        .add("col1", DecimalType.USER_DEFAULT)
+        .add("col2", new DecimalType(38, 38))
+        .add("col3", new DecimalType(38, 0))
+        .add("col4", new DecimalType(38, 10))
+        .add("col5", new DecimalType(5, 2)),
+      TestRow(
+        new JBigDecimal(0),
+        new JBigDecimal("0.01234567891234567891234567891234567890"),
+        new JBigDecimal("123456789123456789123456789123456789"),
+        new JBigDecimal("1234567891234567891234567891.2345678900"),
+        new JBigDecimal("1.23")
+      )
+    )
+  }
+
+  test("parse date type") {
+    testJsonParserForSingleType(
+      jsonString = """{"col1":"2020-12-31"}""",
+      dataType = DateType.DATE,
+      1,
+      TestRow(18627)
+    )
+  }
+
+  test("parse timestamp type") {
+    testJsonParserForSingleType(
+      jsonString =
+        """
+          |{
+          | "col1":"2050-01-01T00:00:00.000-08:00",
```

Comment:
what is the returned value for this? Is it converted into UTC?

Reply:
internal format for timestamp so yes micros since epoch UTC